controllers/progressiverollout_controller.go

```diff
@@ -71,67 +72,117 @@ func (r *ProgressiveRolloutReconciler) Reconcile(ctx context.Context, req ctrl.R
 		log = r.Log.WithValues("stage", stage.Name)
 
 		// Get the clusters to update
-		clusters, err := r.getClustersFromSelector(stage.Targets.Clusters.Selector)
-		if err != nil {
-			log.Error(err, "unable to fetch clusters")
-			return ctrl.Result{}, err
+		clusters, cErr := r.getClustersFromSelector(stage.Targets.Clusters.Selector)
```

Comment:
I'm not particularly fond of these `cErr`, `aErr`, etc. I'd be tempted to keep them in for now and refactor later, as I feel the reconciliation method may need significant refactoring as it's currently very long and difficult to follow.

Reply:
I agree, the problem is that if you re-use `err` you are overwriting the original value, so I'm happy to get any suggestion on that. 
I agree with you that we need some (heavy) refactor.

Reply:
yeah it's common practice to keep and overwrite `err` every time.
do we need to keep `cErr`, `aErr` and `rErr` for a particular reason?

Reply:
https://play.golang.org/p/JniiUIquyoL
`err` doesn't leave the block where it's defined, if it's assigned in the `if` statement.

Reply:
I _think_ it's addressed by https://github.com/Skyscanner/argocd-progressive-rollout/pull/60/commits/f8328b614002fdc58bb334da165ce65f35bc947e.